src/moq/transport.cpp

```diff
@@ -914,6 +914,51 @@ namespace moq {
         }
     }
 
+    void Transport::OnConnectionMetricsSampled(const TimeStampUs sample_time,
+                                               const TransportConnId conn_id,
+                                               QuicConnectionMetrics quic_connection_metrics)
+    {
+        // TODO: doesn't require lock right now, but might need to add lock
+        auto& conn = connections_[conn_id];
+
+        conn.metrics.last_sample_time = sample_time;
+        conn.metrics.quic = quic_connection_metrics;
+
+        if (client_mode_) {
+            MetricsSampled(conn.metrics);
+        } else {
+            MetricsSampled(conn_id, conn.metrics);
```

Comment:
Feels awkward, but it's fine for now.

Reply:
we could call them both, but only one is needed depending on the mode.  We do this for some of the other callbacks based on how they are implemented in the parent class.